erdjs/src/smartcontracts/interaction/interactor.ts

```diff
@@ -0,0 +1,41 @@
+import { ContractFunction } from "../function";
+import { SmartContract } from "../smartContract";
+import { Interaction } from "./interaction";
+import { IInteractionRunner } from "./interface";
+import { TypedValue } from "../typesystem";
+
+export class SmartContractInteractor {
+    private readonly contract: SmartContract;
+    private readonly runner: IInteractionRunner;
+    private preparators: any = {};
+
+    constructor(contract: SmartContract, runner: IInteractionRunner) {
+        this.contract = contract;
+        this.runner = runner;
+
+        this.setupPreparators();
+    }
+
+    private setupPreparators() {
+        let self = this;
+        let abi = this.contract.getAbi();
+
+        for (const definition of abi.getAllEndpoints()) {
+            let functionName = definition.name;
+
+            this.preparators[functionName] = function (args: TypedValue[]) {
+                return self.doPrepare(functionName, args || []);
+            };
+        }
+    }
+
+    prepare(): any {
```

Comment:
Preparators sounds a bit off. In any case, following my previous comment, to ease the developer's life I think we can attach the functions to the main SC object, right?

Reply:
That's correct, we can attach to the `contract` object, will do that.

Reply:
Fixed, we're now using `contract.methods` (inspired from web3js).